code/modules/mob/living/living.dm

```diff
@@ -652,7 +652,8 @@
 		var/mob/living/carbon/CM = L
 		if(CM.on_fire && CM.canmove)
 			CM.fire_stacks -= 5
-			CM.Weaken(3)
+			CM.weakened = max(CM.weakened, 3)//We dont check for CANWEAKEN, I don't care how immune to weakening you are, if you're rolling on the ground, you're busy.
+			CM.update_canmove()

```

Comment:
Can you add an argument to `Weaken` to do this instead of making it happen manually? Something like `/mob/proc/Weaken(amount, forced)`. And skip the `CANWEAKEN` check when `forced`.